src/interfaces/IMiddlewareOptions.ts

```diff
@@ -1,11 +1,16 @@
 import { ClassType } from "class-transformer-validator";
 
 export interface IMiddlewareOptions<T extends object> {
-  classType: ClassType<T>;
+  classType?: ClassType<T>;
+  bodyClassType?: ClassType<T>;
+  queryClassType?: ClassType<T>;
```

Comment:
This would also accept e.g. `{}`, as all properties are optional. Instead, let's use a union type that allows for either only `classType` being set (`IDeprecatedMiddlewareOptions`), only `bodyClassType` being set, only `queryClassType` being set, or both `bodyClassType` and `queryClassType` being set.